### PR TITLE
fix(push): handle deleted environments on floxhub

### PR DESCRIPTION
**Pull request by ysndr**
`flox push` will fetch the latest changes from floxhub prior to pushing to detect diverged branches.
If the environment was removed upstream, the fetch failed. As a result, environments that were deleted before can not be pushed again. With this PR flox will ignore when git is unable to fetch the upstream branch due to it missing and will print a warning instead.

Since this routine is a an internal library detail, its in turn an internal non-user focused warning. Alternatively the warning could be silenced further or implemented as a "precheck" that is called from the frontend crate.

### CODE REVIEW

**Review by mkenigs — COMMENTED** [review]
Not sure this is how we want to display things to users? ping @jennymahmoudi @ghudgins 
```
> flox push
⠂ Pushing updates to FloxHub...
2024-06-13T14:54:01.649127Z  WARN flox_rust_sdk::models::environment::managed_environment: Upstream environment was deleted.
✅ Updates to tmp successfully pushed to FloxHub
```

**Comment by ysndr** [comment]
yeah is a bit wordy of a message, not quite polished

but i'm also not sure if silently pushing is correct either

**Comment by zmitchell** [comment]
I agree that this shouldn't be a `tracing::warn!`, that's for logging and this is a user-facing message. I think it should just be a `messages::warning` so you get the yellow caution emoji.

File: cli/flox-rust-sdk/src/models/environment/managed_environment.rs [hunk 1]
```diff
@@ -1063,10 +1064,17 @@ impl ManagedEnvironment {
...
-        self.floxmeta
+        match self
+            .floxmeta
             .git
             .fetch_ref("dynamicorigin", &format!("+{sync_branch}:{sync_branch}",))
-            .map_err(ManagedEnvironmentError::FetchUpdates)?;
+        {
+            Ok(_) => {},
+            Err(GitRemoteCommandError::RefNotFound(_)) => {
+                warn!("Upstream environment was deleted.")
```
Comment (zmitchell) [inline cli/flox-rust-sdk/src/models/environment/managed_environment.rs:1074]:
```suggestion
                messages::warning("Upstream environment was deleted.")
```

**Review by zmitchell — CHANGES_REQUESTED** [review]

File: cli/flox-rust-sdk/src/models/environment/managed_environment.rs (hunk 1 above)
Comment (ysndr) [reply cli/flox-rust-sdk/src/models/environment/managed_environment.rs:1074]: well messaging lives in the frontend crate and isnt available here
if we want to warn explicitly and user facing we should split that out and precheck earlier or return a status..

Comment (zmitchell) [reply cli/flox-rust-sdk/src/models/environment/managed_environment.rs:1074]: Ok, then if you want to show the user do that, and if not make it a `debug` because it doesn't look good to have logs leaking out to the user.

**Comment by ghudgins** [comment]
What about: ⚠️ Upstream environment was deleted. Attempting to restore.

**Comment by jennymahmoudi** [comment]
> What about: ⚠️ Upstream environment was deleted. Attempting to restore.

Sounds good, if it's followed by the result like `✅ Updates to tmp successfully pushed to FloxHub`.

**Review by dcarley — APPROVED** [review]
Contrary opinion: I wouldn't care to know that the environment was previously deleted, so long as my push worked. I don't recall other tools like `git push` and `docker push` calling these out as warnings.

**Comment by mkenigs** [comment]
Followup in https://github.com/flox/flox/issues/1633
